src/components/dev-hub/modal.js

```diff
@@ -112,15 +114,18 @@ export const Modal = ({
     const [isActive, setIsActive] = useState(isOpenToStart);
     const isMobile = useMedia(screenSize.upToMedium);
     const activateModal = () => setIsActive(true);
-    const deactivateModal = () => setIsActive(false);
+    const deactivateModal = () => {
+        onCloseModal && onCloseModal();
+        setIsActive(false);
+    }
     const closeModalOnEnter = e => {
         const enterKey = 13;
         if (e.keyCode === enterKey) {
             deactivateModal();
         }
     };
     const theme = useTheme();
-    const ResponsiveModal = () =>
+    const responsiveModal = () =>
```

Comment:
It should help to stop creating ResponsiveModal for every render.

Reply:
ah I see, so we kept creating a new modal for each render? I'm assuming this made the callback fire more often than we wanted.